### PR TITLE
DOC: Fixing EX01 - Added examples

### PR DESCRIPTION
**Pull request by DeaMariaLeon**
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

Towards https://github.com/pandas-dev/pandas/issues/37875

Partially fixing https://github.com/pandas-dev/pandas/issues/53668

### CODE REVIEW

File: pandas/_libs/tslibs/timestamps.pyx [hunk 1]
```diff
@@ -1531,6 +1531,72 @@ class Timestamp(_Timestamp):
...
+        try:
+            _dt = datetime(self.year, self.month, self.day,
+                           self.hour, self.minute, self.second,
+                           self.microsecond, self.tzinfo, fold=self.fold)
+        except ValueError as err:
+            raise NotImplementedError(
+                "isocalendar not yet supported on Timestamps which "
+                "are outside the range of Python's standard library. "
+                "For now, please call the components you need (such as `.year` "
+                "and `.month`) and construct your string from there."
```
Comment (MarcoGorelli) [inline pandas/_libs/tslibs/timestamps.pyx:1596]: I don't think this message is quite right, as users aren't constructing a string here

I think it's fine to leave out this part of the message here

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
thanks @DeaMariaLeon !

Would be good to add a little test for these - could just add them to

https://github.com/pandas-dev/pandas/blob/cbad73ac9d68ca2595bb98e1979892545c7b2c11/pandas/tests/scalar/timestamp/test_timestamp.py#L1123-L1132

File: pandas/_libs/tslibs/timestamps.pyx (hunk 1 above)
Comment (DeaMariaLeon) [reply pandas/_libs/tslibs/timestamps.pyx]: Done

File: pandas/_libs/tslibs/timestamps.pyx [hunk 2]
```diff
@@ -2384,10 +2448,27 @@ default 'raise'
...
         # same as super().isoweekday(), but that breaks because of how
         #  we have overridden year, see note in create_timestamp_from_ts
-        return self.weekday() + 1
+        try:
+            _dt = datetime(self.year, self.month, self.day,
+                           self.hour, self.minute, self.second,
+                           self.microsecond, self.tzinfo, fold=self.fold)
+        except ValueError as err:
+            raise NotImplementedError(
+                "isocalendar not yet supported on Timestamps which "
```
Comment (MarcoGorelli) [inline pandas/_libs/tslibs/timestamps.pyx:2468]: should this one say `isoweekday` rather than `isocalendar`?

Comment (DeaMariaLeon) [reply pandas/_libs/tslibs/timestamps.pyx]: Oops. Nice catch. Sorry.

File: pandas/_libs/tslibs/timestamps.pyx [hunk 3]
```diff
@@ -1531,6 +1531,70 @@ class Timestamp(_Timestamp):
...
+        try:
+            _dt = datetime(self.year, self.month, self.day,
+                           self.hour, self.minute, self.second,
+                           self.microsecond, self.tzinfo, fold=self.fold)
+        except ValueError as err:
+            raise NotImplementedError(
+                "date not yet supported on Timestamps which "
+                "are outside the range of Python's standard library. "
+                "For now, please call the components you need (such as `.year` "
+                "and `.month`) and construct your string from there."
```
Comment (MarcoGorelli) [inline pandas/_libs/tslibs/timestamps.pyx:1555]: we're not constructing a string, so we can remove this part of the message

File: pandas/_libs/tslibs/timestamps.pyx [hunk 4]
```diff
@@ -1531,6 +1531,70 @@ class Timestamp(_Timestamp):
...
+            ) from err
+        return _dt.date()
+
+    def dst(self):
+        """
+        Return self.tzinfo.dst(self).
```
Comment (MarcoGorelli) [inline pandas/_libs/tslibs/timestamps.pyx:1561]: How about
```
Return the daylight saving time (DST) adjustment.
```

File: pandas/_libs/tslibs/timestamps.pyx [hunk 5]
```diff
@@ -1531,6 +1531,70 @@ class Timestamp(_Timestamp):
...
+
+        Examples
+        --------
+        >>> ts = pd.Timestamp('2000-06-01 00:00:00',
+        ...                   tz='Europe/Brussels').dst()
+        >>> ts
+        datetime.timedelta(seconds=3600)
```
Comment (MarcoGorelli) [inline pandas/_libs/tslibs/timestamps.pyx:1568]: `ts` usually means "timestamp", so maybe let's do
```python
>>> ts = pd.Timestamp('2000-06-01 00:00:00', tz='Europe/Brussels')
>>> ts.dst()
datetime.timedelta(seconds=3600)
```

File: pandas/_libs/tslibs/timestamps.pyx [hunk 6]
```diff
@@ -1531,6 +1531,70 @@ class Timestamp(_Timestamp):
...
+        """
+        _dt = datetime(self.year, self.month, self.day,
+                       self.hour, self.minute, self.second,
+                       self.microsecond, self.tzinfo, fold=self.fold)
+        return _dt.dst()
```
Comment (MarcoGorelli) [inline pandas/_libs/tslibs/timestamps.pyx:1573]: looks like we're not catching anything here, so can we just return `return super().dst()`?

maybe we can do the same for the others where we're just overriding the docstrings, like
```python
       try:
            return super().isocalendar()
        except ValueError as err:
            raise NotImplementedError(
                "isocalendar not yet supported on Timestamps which "
                "are outside the range of Python's standard library. "
            ) from err
```
and similarly for the others

(in a separate PR, we could also try to do the same trick for the other items where we're overriding the docstrings, like strftime - admittedly I should've thought of that when I made that PR)

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
really sorry, noticed a few more things, so I'm "requesting changes" again

we should also add a whatsnew note to the 2.1.0.rst bug fixes section, noting that `Timestamp.date` was returning incorrect results for inputs outside those supported by the Python standard library's datetime module

File: pandas/_libs/tslibs/timestamps.pyx [hunk 7]
```diff
@@ -1531,6 +1531,70 @@ class Timestamp(_Timestamp):
...
+        >>> ts = pd.Timestamp('2023-01-01 10:00:00.00')
+        >>> ts
+        Timestamp('2023-01-01 10:00:00')
+        >>> ts.date()
+        datetime.date(2023, 1, 1)
+        """
+        try:
+            _dt = datetime(self.year, self.month, self.day,
+                           self.hour, self.minute, self.second,
+                           self.microsecond, self.tzinfo, fold=self.fold)
```
Comment (MarcoGorelli) [inline pandas/_libs/tslibs/timestamps.pyx:1549]: can we construct a `date` object directly? like
```python
_dt = date(self.year, self.month, self.day)
```
, where `date` is imported from `datetime`?

Comment (MarcoGorelli) [reply pandas/_libs/tslibs/timestamps.pyx:1549]: might work to do `import datetime as dt`, then use `dt.date`

File: pandas/_libs/tslibs/timestamps.pyx [hunk 8]
```diff
@@ -2384,10 +2444,27 @@ default 'raise'
...
+        except ValueError as err:
+            raise NotImplementedError(
+                "isoweekday not yet supported on Timestamps which "
+                "are outside the range of Python's standard library. "
+            ) from err
+        return _dt.weekday() + 1
```
Comment (MarcoGorelli) [inline pandas/_libs/tslibs/timestamps.pyx:2467]: wait sorry, why did this one have to change? isn't `self.weekday() + 1` correct?

File: pandas/tests/scalar/timestamp/test_timestamp.py [hunk 9]
```diff
@@ -1124,9 +1124,27 @@ def test_negative_dates():
...
+    )
+    func = "^date"
+    with pytest.raises(NotImplementedError, match=func + msg):
+        ts.date()
+    func = "^isocalendar"
+    with pytest.raises(NotImplementedError, match=func + msg):
+        ts.isocalendar()
+    func = "^isoweekday"
+    with pytest.raises(NotImplementedError, match=func + msg):
+        ts.isoweekday()
```
Comment (MarcoGorelli) [inline pandas/tests/scalar/timestamp/test_timestamp.py:1147]: see my comment above, not sure this one needs to raise?

File: doc/source/whatsnew/v2.1.0.rst [hunk 10]
```diff
@@ -350,6 +350,7 @@ Datetimelike
 ^^^^^^^^^^^^
 - :meth:`DatetimeIndex.map` with ``na_action="ignore"`` now works as expected. (:issue:`51644`)
 - Bug in :func:`date_range` when ``freq`` was a :class:`DateOffset` with ``nanoseconds`` (:issue:`46877`)
+- Bug in :meth:`Timestamp.date`, :meth:`Timestamp.isocalendar`, and :meth:`Timestamp.isoweekday` were returning incorrect results for inputs outside those supported by the Python standard library's datetime module (:issue:`53668`)
```
Comment (MarcoGorelli) [inline doc/source/whatsnew/v2.1.0.rst:353]: I think we can remove `isoweekday`

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
sorry one more comment (in any case, CI needs fixing, but hopefully #53730 and
#53727 will do that)

File: pandas/_libs/tslibs/timestamps.pyx (hunk 8 above)
Comment (DeaMariaLeon) [reply pandas/_libs/tslibs/timestamps.pyx:2467]: I thought that if we give it a negative date it should raise. So, in the stable version:
for `ts = pd.Timestamp('-2000')` the output is:
`ts.isoweekday()=7`
`ts.weekday()=6`

File: pandas/tests/scalar/timestamp/test_timestamp.py (hunk 9 above)
Comment (DeaMariaLeon) [reply pandas/tests/scalar/timestamp/test_timestamp.py:1147]: Please see my question above.

File: doc/source/whatsnew/v2.1.0.rst (hunk 10 above)
Comment (DeaMariaLeon) [reply doc/source/whatsnew/v2.1.0.rst]: Requested changes done.

**Review by MarcoGorelli — APPROVED** [review]
nice, should be good now - thanks @DeaMariaLeon !

**Pull request merged**